.github/workflows/multinode.yml

```diff
@@ -0,0 +1,364 @@
+          multinode_keypair    = "${{ env.MULTINODE_NAME }}"
+          multinode_vm_network = "${{ env.MULTINODE_NETWORK }}"
+          multinode_vm_subnet  = "${{ env.MULTINODE_SUBNET }}"
+          compute_count        = "${{ env.MULTINODE_COMPUTE_COUNT }}"
+          controller_count     = "${{ env.MULTINODE_CONTROLLER_COUNT }}"
+          compute_disk_size    = 100
+          controller_disk_size = 100
+
+          ssh_public_key = "id_rsa.pub"
+          ssh_user       = "${{ env.SSH_USERNAME }}"
+
+          storage_count     = "${{ env.MULTINODE_STORAGE_COUNT }}"
+          storage_flavor    = "${{ env.MULTINODE_STORAGE_FLAVOR }}"
+          storage_disk_size = 100
+
+          deploy_wazuh       = false
+          infra_vm_flavor    = "${{ env.MULTINODE_INFRA_VM_FLAVOR }}"
+          infra_vm_disk_size = 100
+          EOF
+
+          if [[ "${{ inputs.ssh_key }}" != "" ]]; then
+            cat << EOF >> terraform.tfvars
+          add_ansible_control_fip = true
+          ansible_control_fip_pool = "${{ env.MULTINODE_FIP_POOL }}"
+          EOF
+          fi
+        working-directory: ${{ github.workspace }}/terraform-kayobe-multinode
+        env:
+          MULTINODE_NAME: "${{ inputs.multinode_name }}"
+          MULTINODE_ANSIBLE_CONTROL_VM_FLAVOR: ${{ vars.multinode_ansible_control_vm_flavor }} # en1.xsmall
+          MULTINODE_SEED_VM_FLAVOR: ${{ vars.multinode_seed_vm_flavor }} # en1.xsmall
+          MULTINODE_INFRA_VM_FLAVOR: ${{ vars.multinode_infra_vm_flavor }} # en1.xsmall
+          MULTINODE_FLAVOR: ${{ vars.multinode_flavor }} # en1.large
+          MULTINODE_STORAGE_FLAVOR: ${{ vars.multinode_storage_flavor }} # en1.medium
+          MULTINODE_COMPUTE_COUNT: "${{ inputs.multinode_compute_count }}"
+          MULTINODE_CONTROLLER_COUNT: "${{ inputs.multinode_controller_count }}"
+          MULTINODE_STORAGE_COUNT: "${{ inputs.multinode_storage_count }}"
+          MULTINODE_IMAGE: ${{ steps.image_name.outputs.image_name }}
+          MULTINODE_NETWORK: ${{ vars.multinode_network }}
+          MULTINODE_SUBNET: ${{ vars.multinode_subnet }}
+          MULTINODE_FIP_POOL: ${{ vars.multinode_fip_pool }}
+          SSH_USERNAME: "${{ inputs.ssh_username }}"
+
+      - name: Initialise terraform
+        run: terraform init
+        working-directory: ${{ github.workspace }}/terraform-kayobe-multinode
+
+      - name: Validate terraform
+        id: tf_validate
+        run: terraform validate
+        working-directory: ${{ github.workspace }}/terraform-kayobe-multinode
+
+      - name: Configure Ansible
+        run: |
+          echo '${{ env.KAYOBE_VAULT_PASSWORD }}' > vault-pw
+
+          cat << EOF >> ansible/vars/defaults.yml
+          kayobe_config_version: ${{ inputs.upgrade && inputs.stackhpc_kayobe_config_previous_version || inputs.stackhpc_kayobe_config_version }}
+          ssh_key_path: ${{ github.workspace }}/terraform-kayobe-multinode/id_rsa
+          vxlan_vni: 53299
```

Comment:
Would be nice if this was randomised/customisable, perhaps in some bounded range for "CI multinodes"